morango/crypto.py

```diff
@@ -168,18 +114,170 @@ def _add_pem_headers(self, pem_string, header_string):
         }
         return "-----BEGIN %(header_string)s-----\n%(key)s\n-----END %(header_string)s-----" % context
 
+    def ensure_bytes(self, message):
+        try:
+            return message.encode("utf-8", "replace")
+        except (UnicodeDecodeError, TypeError, AttributeError):
+            return message
+
     def __str__(self):
         return self.get_public_key_string()
 
 
-def hashed(message):
-    # try to encode the message as UTF-8, replacing any invalid characters so they don't blow up the hashing
-    try:
-        message = message.encode("utf-8", "replace")
-    except UnicodeDecodeError:
-        pass
-    return hashlib.sha256(message).digest()
-    
+class PythonRSAKey(BaseKey):
+
+    _public_key = None
+    _private_key = None
```

Comment:
I did originally, then realized they were only being referenced in the child classes, and were more of an internal implementation detail, as not all "backends" (subclasses) might implement this in the same way (e.g. some might have a single internal key object that stores both the public and private components).